### PR TITLE
Create Tests for EditBookCommandParser and Fix bugs for EditBookCommand

### PR DESCRIPTION
**Pull request by yuqitanyq**
**Create tests for EditBookCommandParser and fix bugs for EditBookCommand**

Notable changes:
Created a Test class with tests specifically for EditBookCommandParser
Fixed display bug for EditBookCommand

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/book/EditBookCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    }
+
+    @Test
+    public void parse_invalidPreamble_failure() {
+        // negative index
+        assertParseFailure(parser, "-5" + BOOK_NAME_DESC_HARRY_POTTER, MESSAGE_INVALID_FORMAT);
+
+        // zero index
+        assertParseFailure(parser, "0" + BOOK_NAME_DESC_HUNGER_GAMES, MESSAGE_INVALID_FORMAT);
+
```
Comment (yl-ang) [inline src/test/java/seedu/address/logic/parser/book/EditBookCommandParserTest.java:70]: Good boundary value analysis test cases!

File: src/test/java/seedu/address/logic/parser/book/EditBookCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        EditBookCommand.EditBookDescriptor bookDescriptor = new EditBookDescriptorBuilder()
+                .withBookName(VALID_BOOK_NAME_HUNGER_GAMES)
+                .withIsbn(VALID_ISBN_HUNGER_GAMES)
+                .withAuthors(VALID_AUTHOR_SUZANNE_COLLINS)
+                .withTags(VALID_TAG_THRILLER, VALID_TAG_SCIFI)
+                .build();
+        EditBookCommand expectedCommand = new EditBookCommand(targetIndex, bookDescriptor);
+
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
```
Comment (yl-ang) [inline src/test/java/seedu/address/logic/parser/book/EditBookCommandParserTest.java:189]: Good test case to check if the parser handles repeated fields as excepted :100:

File: src/test/java/seedu/address/logic/parser/book/EditBookCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+        // invalid tag
+        assertParseFailure(parser, "1" + INVALID_TAG_DESC, Tag.MESSAGE_CONSTRAINTS);
+
+        // invalid isbn followed by valid author
+        assertParseFailure(parser, "1" + INVALID_ISBN_DESC + AUTHOR_DESC_JK_ROWLING, Isbn.MESSAGE_CONSTRAINTS);
+
+        // valid isbn followed by invalid isbn. The test case for invalid isbn followed by valid isbn
+        // is tested at {@code parse_invalidValueFollowedByValidValue_success()}
+        assertParseFailure(parser, "1" + ISBN_DESC_HARRY_POTTER + INVALID_ISBN_DESC, Isbn.MESSAGE_CONSTRAINTS);
```
Comment (yl-ang) [inline src/test/java/seedu/address/logic/parser/book/EditBookCommandParserTest.java:97]: Good edge case test!

**Review by yl-ang — APPROVED** [review]
LGTM! Test cases added are relevant and help to test the edit book command parser. In addition, the test cases helped to increase our overall code coverage by 3%, good job @yuqitanyq :100:
